retrofit-adapters/rxjava2/src/main/java/retrofit2/adapter/rxjava2/NoContentException.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2.adapter.rxjava2;
+
+import io.reactivex.Completable;
+import retrofit2.Response;
+
+/** Exception for an unexpected, 204/205 HTTP response when not used with {@link Completable}. */
+public final class NoContentException extends RuntimeException {
```

Comment:
Does this type need to exist? Seems like a normal `HttpException` could be used.

Reply:
We could! The doc for `HttpException` currently says it's only for `non-2xx HTTP response`, so I'm not sure if this would constitute an undesirable behavior change

Reply:
Yeah I suspect it might. We wouldn't even want to subclass either because an `instanceof` check still might be happening. Let's leave it separate.